File: Python/Algorithms/BFS.py
from queue import Queue

# Basic graph implementation for reference
class Graph:
    def __init__(self, size):
        self.adj = [[]] * size
        self.size = size

# BFS method that accepts a Graph and start vertex 
def bfs(G, startVert):
    
    visited = [False] * G.size
    q = Queue()

    visited[startVert] = True

    q.put(startVert)

    while not q.empty():
        v = q.get()
        for adjV in G.adj[v]:
            if not visited[adjV]:
                visited[adjV] = True
                q.put(adjV)

File: CPlusPlus/Algorithms/BFS.cpp
#include <list>

// Example class of a graph implementation:
class Graph {
    public:
        Graph(int Size) 
            :Size(Size) {
            adj = new std::list<int>[Size];
        }
        ~Graph()
        {
            delete[] adj;
        }
        std::list<int>* adj;
        int Size;
};


void bfs(Graph & G, const int startVert) {

    bool* visited = new bool[G.Size];
    for(int i = 0; i < G.Size; ++i) {
        visited[i] = false;
    }

    std::list<int> queue;

    // Mark and enqueue our start vert
    visited[startVert] = true;
    queue.push_back(startVert);

    while(!queue.empty()) {
        // Get the next vert
        int v = queue.front();
        queue.pop_front();
        // Explore it:
        std::list<int>::iterator it;
        for(it = G.adj[v].begin(); it != G.adj[v].end(); ++it) {

            if(!visited[*it]) {
                visited[*it] = true;
                queue.push_back(*it);
            }

        }
    }
    delete[] visited;
}